src/extension/background-script/actions/nostr/helpers.ts

```diff
@@ -45,13 +45,18 @@ export async function addPermissionFor(method: string, host: string) {
   return !!permissionIsAdded && (await db.saveToStorage());
 }
 
-export function validateEvent(event: Event) {
-  if (event.id !== getEventHash(event)) return false;
+// from: https://github.com/nbd-wtf/nostr-tools/blob/160987472fd4922dd80c75648ca8939dd2d96cc0/event.ts#L61
+// to avoid the additional dependency
+export function validateEvent(event: Event): boolean {
   if (typeof event.content !== "string") return false;
   if (typeof event.created_at !== "number") return false;
+  // ignore these checks because if the pubkey is not set we add it to the event. same for the ID.
+  // if (typeof event.pubkey !== "string") return false;
+  // if (!event.pubkey.match(/^[a-f0-9]{64}$/)) return false;
 
   if (!Array.isArray(event.tags)) return false;
-  for (const tag of event.tags) {
+  for (let i = 0; i < event.tags.length; i++) {
```

Comment:
nit: can `forEach` bs used?: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

Reply:
this is from: https://github.com/nbd-wtf/nostr-tools/blob/master/event.ts#L61-L77
I copied it from there to be in sync.